src/gui/qgsmapcanvas.cpp

```diff
@@ -147,6 +147,8 @@ QgsMapCanvas::QgsMapCanvas( QWidget *parent )
   mSettings.setFlag( QgsMapSettings::DrawEditingInfo );
   mSettings.setFlag( QgsMapSettings::UseRenderingOptimization );
   mSettings.setFlag( QgsMapSettings::RenderPartialOutput );
+  mSettings.setDestinationCrs( QgsProject::instance()->crs() );
+  mSettings.setEllipsoid( QgsProject::instance()->ellipsoid() );
```

Comment:
OK, added in https://github.com/qgis/QGIS/commit/dd6c7fdbc8b57a606c00e26a09c8884774122e68